project-reports-lib.ts

```diff
@@ -75,6 +76,58 @@ export function getStringFromLabel(card: ProjectIssue, re: RegExp): string {
   return str
 }
 
+export function getLastCommentField(
+  issue: ProjectIssue,
+  field: string
+): string {
+  let val = ''
+
+  if (!issue.comments) {
+    return ''
+  }
+
+  for (let i = issue.comments.length - 1; i >= 0; i--) {
+    const comment = issue.comments[i]
+    if (!comment) {
+      break
+    }
+
+    const lines = comment.body.split(os.EOL)
+    for (const line of lines) {
+      const parts = line.trim().split(':')
+      if (parts.length === 2 && fuzzyMatch(parts[0], field)) {
+        val = parts[1].trim()
+        break
+      }
+    }
+
+    if (val) {
+      break
+    }
+  }
+
+  return val
+}
+
+// returns a valid date field value from a comment field
+export function getLastCommentDateField(
+  issue: ProjectIssue,
+  field: string
+): Date {
+  let d: Date = null
+  const val = getLastCommentField(issue, field)
+
+  try {
+    if (val) {
+      d = new Date(val)
```

Comment:
`new Date()` won't throw with an invalid input—instead, it'll either return `Invalid Date` later when it's cast to a string, or it may throw if something like `toISOString()` is called on it.

Instead, an alternative would be to call `Date.parse(val)` and then check `isNan(date)`. You can then return null if the parsed date is NaN, or you can return `new Date(parsedDate)`.

Even safer might instead be to require that the date is in a format that we know passes `new Date` using a regular expression such as `/^\d{4}-\d{2}-\d{2}$/`.

Reply:
I took out the try catch and validated you get invalid date which if json stringified is null